riscv/processor.cc

```diff
@@ -886,6 +889,14 @@ void processor_t::take_trap(trap_t& t, reg_t epc)
     vsdeleg = (curr_virt && state.prv <= PRV_S) ? (state.medeleg->read() & state.hedeleg->read()) : 0;
     hsdeleg = (state.prv <= PRV_S) ? state.medeleg->read() : 0;
   }
+  // An unexpected trap - a trap when SDT is 1 - traps to M-mode
+  if ( (state.prv <= PRV_S && bit < max_xlen) &&
+       (((vsdeleg >> bit) & 1)  || ((hsdeleg >> bit) & 1)) ) {
```

Comment:
Minor nit, in this project we prefer not to add a space after the opening `(` or below the closing `)`.